modules/scope3/rtd/module.go

```diff
@@ -0,0 +1,401 @@
+				Activities: []hookanalytics.Activity{{
+					Name:   "scope3_fetch",
+					Status: hookanalytics.ActivityStatusError,
+					Results: []hookanalytics.Result{{
+						Status: hookanalytics.ResultStatusError,
+						Values: map[string]interface{}{"error": err.Error()},
+					}},
+				}},
+			},
+		}, nil
+	}
+
+	// Store segments in module context
+	if segmentStore, ok := miCtx.ModuleContext["segments"].(*sync.Map); ok {
+		segmentStore.Store("segments", segments)
+	}
+
+	// Store segments for later use - no mutation needed at this stage
+	changeSet := hookstage.ChangeSet[hookstage.RawAuctionRequestPayload]{}
+
+	return hookstage.HookResult[hookstage.RawAuctionRequestPayload]{
+		ChangeSet: changeSet,
+		AnalyticsTags: hookanalytics.Analytics{
+			Activities: []hookanalytics.Activity{{
+				Name:   "scope3_fetch",
+				Status: hookanalytics.ActivityStatusSuccess,
+				Results: []hookanalytics.Result{{
+					Status: hookanalytics.ResultStatusModify,
+					Values: map[string]interface{}{
+						"segments": segments,
+						"count":    len(segments),
+					},
+				}},
+			}},
+		},
+	}, nil
+}
+
+// HandleAuctionResponseHook adds targeting data to the auction response
+func (m *Module) HandleAuctionResponseHook(
+	ctx context.Context,
+	miCtx hookstage.ModuleInvocationContext,
+	payload hookstage.AuctionResponsePayload,
+) (hookstage.HookResult[hookstage.AuctionResponsePayload], error) {
+	// Retrieve segments from module context
+	var segments []string
+	if segmentStore, ok := miCtx.ModuleContext["segments"].(*sync.Map); ok {
+		if val, ok := segmentStore.Load("segments"); ok {
+			segments = val.([]string)
+		}
+	}
+
+	if len(segments) == 0 {
+		return hookstage.HookResult[hookstage.AuctionResponsePayload]{}, nil
+	}
+
+	// Add segments to the auction response
+	changeSet := hookstage.ChangeSet[hookstage.AuctionResponsePayload]{}
+	changeSet.AddMutation(
+		func(payload hookstage.AuctionResponsePayload) (hookstage.AuctionResponsePayload, error) {
+			// Add Scope3 segments to the response ext so publisher can use them
+			if payload.BidResponse.Ext == nil {
+				payload.BidResponse.Ext = json.RawMessage("{}")
+			}
+
+			var extMap map[string]interface{}
+			if err := jsonutil.Unmarshal(payload.BidResponse.Ext, &extMap); err != nil {
+				extMap = make(map[string]interface{})
+			}
+
+			// Add segments as individual targeting keys for GAM integration
+			if m.cfg.AddToTargeting {
+				if prebidMap, ok := extMap["prebid"].(map[string]interface{}); ok {
+					if targetingMap, ok := prebidMap["targeting"].(map[string]interface{}); ok {
+						// Add each segment as individual targeting key
+						for _, segment := range segments {
+							targetingMap[segment] = "true"
+						}
+					} else {
+						// Create targeting map with individual segment keys
+						newTargeting := make(map[string]interface{})
+						for _, segment := range segments {
+							newTargeting[segment] = "true"
+						}
+						prebidMap["targeting"] = newTargeting
+					}
```

Comment:
Can you add test coverage for these blocks?

Reply:
Added a couple of tests for this